### ISSUE TITLE
Integrate Next.js

### ISSUE THREAD

**Issue by grez96** [Jam3/react-ui#56]
Use Next.js in react-ui so the Button/Link component can be easier to integrate.

**Comment by jmckinnell** [comment]
I'm of the opinion that we should keep `react-ui` platform agnostic. Button and Link are easy enough to include in the boiler-plate and any style/animation based components can just wrap the button etc.

**Comment by grez96** [comment]
I agree keeping `react-ui` platform agnostic is best. Maybe we can add Next.js as a peer dependency, that way these components can just import ex. `import Link from 'next/link'`. Doing this the components will become super easy plug and play.
Curious why this [PR ](https://github.com/Jam3/react-ui/pull/42/files)that moved react-router to peer dependencies was closed?

Maybe we can also build component variations?
* `Button_NextJS`
* `Button_CRA`
